### PR TITLE
Use customizable prefix key

### LINKED ISSUE #407: Use customizable prefix key
## Expected behavior

The prefix key for `clojure-mode-map` should be customizable, as it is in packages like [Projectile](https://github.com/bbatsov/projectile/blob/master/projectile.el#L202).

## Actual behavior

The `C-c C-r` key sequence is [hard-coded](https://github.com/clojure-emacs/clojure-mode/blob/master/clojure-mode.el#L194).

## Steps to reproduce the problem

Obvious.

## Environment & Version information

Not needed.

### clojure-mode version information

`clojure-mode`: 5.5.2

### Emacs version

Emacs: 24.5.50.1

### Operating system

OS: Fedora 14

### PR DESCRIPTION
**Pull request by xiongtx**
Fixes #407.

-----------------

Before submitting a PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines][1]
- [ ] You've added tests (if possible) to cover your change(s). Indentation & font-lock tests are extremely important!
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] All tests are passing (`make test`)
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [ ] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

[1]: https://github.com/clojure-emacs/clojure-mode/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: clojure-mode.el [hunk 1]
```diff
@@ -191,38 +191,55 @@ Out-of-the box clojure-mode understands lein, boot and gradle."
           (and (listp value)
                (cl-every 'stringp value))))
 
+(defcustom clojure-mode-keymap-prefix "C-c C-r"
```
Comment (bbatsov) [inline clojure-mode.el]: The refactoring prefix shouldn't really be customizable in this manner. A major mode's keybindings should always start with `C-c`, no exceptions. Only the `C-r` part might be made customizable, but I don't see much value in doing this.

File: clojure-mode.el [hunk 2]
```diff
@@ -191,38 +191,55 @@ Out-of-the box clojure-mode understands lein, boot and gradle."
...
+  "Clojure mode keymap prefix."
+  :group 'clojure
+  :type 'string)
+
+(defvar clojure-mode-prefixed-bindings
```
Comment (bbatsov) [inline clojure-mode.el]: I'd define a nested keymap, instead of generating keybindings dynamically.

File: clojure-mode.el (hunk 1 above)
Comment (Malabarba) [inline clojure-mode.el]: Also, it should be called `clojure-mode-refactorings-keymap-prefix` or something like that. As only our refactoring commands are under this prefix.

**Comment by xiongtx** [comment]
Closed due to [not being needed](https://github.com/clojure-emacs/clojure-mode/issues/407#issuecomment-247744686).

**Pull request closed without merging**